version: '3.8'

services:

  none:  # No database engine configured
    build:
      context: .
      args:
        ENVIRONMENT: development
    command: pytest
    volumes:
      - .:/code

  django:
    build:
      context: .
      args:
        ENVIRONMENT: development
        DJANGO_VERSION: 4.2.0
    command: pytest
    volumes:
      - .:/code

  sqlalchemy:
    build:
      context: .
      args:
        ENVIRONMENT: development
        SQLALCHEMY_VERSION: 1.3.0
    command: pytest
    volumes:
      - .:/code

  docs:
    build:
      context: .
      dockerfile: docs/Dockerfile
    command: mkdocs serve --dev-addr 0.0.0.0:8008 --watch-theme
    volumes:
      - .:/app
    ports:
      - 8008:8008

  qa:
    build:
      context: ./qa
      args:
        DJANGO_VERSION: 4.2
        ACQUIRING_VERSION: 0.4.1
    volumes:
      - ./qa:/code
    ports:
      - 8888:8888
    depends_on:
      - postgres
    command: python manage.py runserver 0.0.0.0:8888

  postgres:
    image: postgres:16
    environment:
      POSTGRES_HOST_AUTH_METHOD: "trust"
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data

volumes:
  postgres_data:
